### ISSUE TITLE
Error with DFG Miner and filtered DataFrame

### ISSUE THREAD

**Issue by joleaf** [process-intelligence-solutions/pm4py#360]
Hi, first of all, thank you for this great Framework!

I run into a KeyError when I try to run `pm4py.discover_dfg_typed(log)` and fail at this line:
https://github.com/pm4py/pm4py-core/blob/e46b1baa671d480a823c04e0748ef7c0398277c6/pm4py/algo/discovery/dfg/variants/clean.py#L55

What I did to fix this was:
`log = log.reset_index(drop=True)`

However, I don't know if this is a proper fix for my situation.

**Comment by fit-alessandro-berti** [comment]
Dear @joleaf 

We were not able to reconstruct the error. Could you give more insights / an example script where the problem occurs?

The following works fine both starting from XES and CSV logs


import pm4py

log = pm4py.read_xes("tests/input_data/receipt.xes")
dfg = pm4py.discover_dfg_typed(log)

print(dfg)

import pandas as pd

dataframe = pd.read_csv("tests/input_data/receipt.csv")
dfg2 = pm4py.discover_dfg_typed(dataframe)

print(dfg2)

**Comment by joleaf** [comment]
Sry, a little bit more context is necessary:
If I filter the DataFrame (e.g. for a specific case) and thereby remove the first Series (Index 0). The function `discover_dfg_type` will fail at this line, because index 0 is not present in the DataFrame:
https://github.com/pm4py/pm4py-core/blob/e46b1baa671d480a823c04e0748ef7c0398277c6/pm4py/algo/discovery/dfg/variants/clean.py#L55

Example with the `running-example.xes` file:
```python
import pm4py

def mine_dfg(e_log):
    try:
        dfg = pm4py.discover_dfg_typed(log)
    except Exception:
        print("Error mining dfg: KeyError --> Remove the try to see the error")
        return None
    else:
        print("Success mining dfg")
        return dfg

log = pm4py.read_xes("running-example.xes")
print("Unfiltered Log")
print(log.head())
mine_dfg(log)
print()

print("Filtered Log, only case 6")
log = pm4py.filter_trace_attribute_values(log,"case:concept:name", ["6"])
# log = log[log['case:concept:name']=="6"] # does not work either
print(log.head())
mine_dfg(log) # error here
print()

print("Reset index of dataframe")
log = log.reset_index(drop=True)
print(log.head()) # fixed
mine_dfg(log)
```

**Output**:
```text
parsing log, completed traces :: 100%|█████████████████████████████████| 6/6 [00:00<00:00, 3316.97it/s]
Unfiltered Log
         concept:name org:resource            time:timestamp            Activity Resource Costs case:concept:name    case:creator
0    register request         Pete 2010-12-30 14:32:00+01:00    register request     Pete    50                 3  Fluxicon Nitro
1    examine casually         Mike 2010-12-30 15:06:00+01:00    examine casually     Mike   400                 3  Fluxicon Nitro
2        check ticket        Ellen 2010-12-30 16:34:00+01:00        check ticket    Ellen   100                 3  Fluxicon Nitro
3              decide         Sara 2011-01-06 09:18:00+01:00              decide     Sara   200                 3  Fluxicon Nitro
4  reinitiate request         Sara 2011-01-06 12:18:00+01:00  reinitiate request     Sara   200                 3  Fluxicon Nitro
Success mining dfg

Filtered Log, only case 4
        concept:name org:resource            time:timestamp          Activity Resource Costs case:concept:name    case:creator
19  register request         Mike 2011-01-06 15:02:00+01:00  register request     Mike    50                 6  Fluxicon Nitro
20  examine casually        Ellen 2011-01-06 16:06:00+01:00  examine casually    Ellen   400                 6  Fluxicon Nitro
21      check ticket         Mike 2011-01-07 16:22:00+01:00      check ticket     Mike   100                 6  Fluxicon Nitro
22            decide         Sara 2011-01-07 16:52:00+01:00            decide     Sara   200                 6  Fluxicon Nitro
23  pay compensation         Mike 2011-01-16 11:47:00+01:00  pay compensation     Mike   200                 6  Fluxicon Nitro
Error mining dfg: KeyError --> Remove the try to see the error

Reset index of dataframe
       concept:name org:resource            time:timestamp          Activity Resource Costs case:concept:name    case:creator
0  register request         Mike 2011-01-06 15:02:00+01:00  register request     Mike    50                 6  Fluxicon Nitro
1  examine casually        Ellen 2011-01-06 16:06:00+01:00  examine casually    Ellen   400                 6  Fluxicon Nitro
2      check ticket         Mike 2011-01-07 16:22:00+01:00      check ticket     Mike   100                 6  Fluxicon Nitro
3            decide         Sara 2011-01-07 16:52:00+01:00            decide     Sara   200                 6  Fluxicon Nitro
4  pay compensation         Mike 2011-01-16 11:47:00+01:00  pay compensation     Mike   200                 6  Fluxicon Nitro
Success mining dfg

```

**Comment by fit-alessandro-berti** [comment]
Perfectly clear error description.
Thank you!

**Comment by fit-alessandro-berti** [comment]
The issue has been resolved in pm4py 2.3.2

**Comment by joleaf** [comment]
Okay, nice, thank you! However, this might be an error as well if you don't drop the old index with `drop=True`? Just an assumption, I didn't test this.
https://github.com/pm4py/pm4py-core/blob/976bf88c0fb5bcadde2195d8446574463f9c4594/pm4py/algo/discovery/dfg/variants/clean.py#L50
